tests/test_custom_models.py

```diff
@@ -245,6 +246,14 @@
     ("Conv2d 3 OFT", "Conv2d", OFTConfig, {"target_modules": ["conv2d"], "coft": True}),
     ("Conv2d 4 OFT", "Conv2d", OFTConfig, {"target_modules": ["conv2d"], "block_share": True}),
     ("Conv2d 5 OFT", "Conv2d", OFTConfig, {"target_modules": ["conv2d"], "coft": True, "block_share": True}),
+    ########
+    # HRA #
+    ########
+    ("Vanilla MLP 1 HRA", "MLP", HRAConfig, {"target_modules": "lin0"}),
+    ("Vanilla MLP 2 HRA", "MLP", HRAConfig, {"target_modules": ["lin0"]}),
+    ("Vanilla MLP 3 HRA", "MLP", HRAConfig, {"target_modules": ["lin0", "lin1"]}),
```

Comment:
Gram-Schmidt orthogonalization results in strict orthogonality, which prevents an untrained HRA from being initialized as an identity transform. Consequently, many test cases related to initialization would fail. Therefore, I did not add a test with `apply_GS=True`.

Reply:
I see. Still, it would be nice if we could test that, to ensure that this part of the code is well covered. I have a suggestion: Maybe you could add a test to `tests_common_gpu.py` that is similar to this one:

https://github.com/huggingface/peft/blob/7ffa43b16e4625322c9f157db12c9358f2f54cdf/tests/test_common_gpu.py#L946

(but bnb quantization is not needed).

The test could instantiate a causal LM, create the output, then apply PEFT with HRA+GS and check the output again. The two outputs should be different.

If there is something else that could be tested, e.g. by checking that some weights are initialized in a certain way, that could also be added. WDYT?

Reply:
OK, we have added a test case for HRA+GS, as well as a test case to verify that HRA cannot be initialized as an identity transform when r is an odd number.